app/src/main/kotlin/com/camerasecuritysystem/client/gallery/DashcamGalleryFragment.kt

```diff
@@ -0,0 +1,96 @@
+package com.camerasecuritysystem.client.gallery
+
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.media.ThumbnailUtils
+import android.os.Build
+import android.os.Bundle
+import android.os.CancellationSignal
+import android.provider.MediaStore
+import android.util.Log
+import android.util.Size
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.RequiresApi
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.camerasecuritysystem.client.R
+import com.camerasecuritysystem.client.models.Video
+import java.io.File
+import android.media.MediaMetadataRetriever
+import androidx.core.content.FileProvider
+import com.camerasecuritysystem.client.BuildConfig
+import java.util.*
+import kotlin.collections.ArrayList
+
+
+class DashcamGalleryFragment : Fragment() {
+
+    private lateinit var recyclerView: RecyclerView
+    private lateinit var recyclerViewLayoutManager: RecyclerView.LayoutManager
+
+    @RequiresApi(Build.VERSION_CODES.Q)
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        var root = inflater.inflate(R.layout.fragment_dashcam_gallery, container, false)
+
+        recyclerView = root.findViewById(R.id.recyclerView)
+        recyclerViewLayoutManager = GridLayoutManager(context, 3)
+        recyclerView.layoutManager = recyclerViewLayoutManager
+
+        return root
+    }
+
+    @RequiresApi(Build.VERSION_CODES.Q)
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        fetchVideosFromFiles()
+    }
+
+    @RequiresApi(Build.VERSION_CODES.Q)
+    private fun fetchVideosFromFiles() {
+
+        val path = "${requireContext().filesDir}/dashcam/"
+
+        val directory = File(path)
+
+        val files = directory.listFiles()
+        val retriever = MediaMetadataRetriever()
+
+        var videoArray = ArrayList<Video>()
+        var index = 0
```

Comment:
I think we should use the type system as much as possible. Could this be an Int? The same goes for other places where it could probable be used.
```suggestion
        var index: Int = 0
```